lib/error.js

```diff
@@ -54,10 +63,11 @@ export const makeError = ({
 	const execaMessage = `Command ${prefix}: ${command}`;
 	const isError = Object.prototype.toString.call(error) === '[object Error]';
 	const shortMessage = isError ? `${execaMessage}\n${error.message}` : execaMessage;
-	const message = [shortMessage, stdio[2], stdio[1], ...stdio.slice(3)]
+	const messageStdio = all === undefined ? [stdio[2], stdio[1]] : [all];
```

Comment:
I have just added this line too: if `all: true` is used, the error message should show `result.all` instead, since it is intertwined, so it is more useful for debugging that splitting `stdout` and `stderr`.